src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -36,202 +79,565 @@ public SafeX509ChainHandle SafeHandle
+
+            if (baseUri == null)
+            {
+                return status;
+            }
+            
+            using (SafeOcspRequestHandle req = Interop.Crypto.X509ChainBuildOcspRequest(_storeCtx))
+            {
+                ArraySegment<byte> encoded = Interop.Crypto.OpenSslRentEncode(
+                    handle => Interop.Crypto.GetOcspRequestDerSize(handle),
+                    (handle, buf) => Interop.Crypto.EncodeOcspRequest(handle, buf),
+                    req);
+
+                ArraySegment<char> urlEncoded = Base64UrlEncode(encoded);
+                string requestUrl = UrlPathAppend(baseUri, urlEncoded);
+
+                // Nothing sensitive is in the encoded request (it was sent via HTTP-non-S)
+                ArrayPool<byte>.Shared.Return(encoded.Array);
+                ArrayPool<char>.Shared.Return(urlEncoded.Array);
+
+                // https://tools.ietf.org/html/rfc6960#appendix-A describes both a GET and a POST
+                // version of an OCSP responder.
+                //
+                // Doing POST via the reflection indirection to HttpClient is difficult, and
+                // CA/Browser Forum Baseline Requirements (version 1.6.3) section 4.9.10
+                // (On-line REvocation Checking Requirements) says that the GET method must be supported.
+                //
+                // So, for now, only try GET.
+
+                SafeOcspResponseHandle resp =
+                    CertificateAssetDownloader.DownloadOcspGet(requestUrl, ref _remainingDownloadTime);
+
+                using (resp)
+                {
+                    if (resp == null || resp.IsInvalid)
+                    {
+                        return Interop.Crypto.X509VerifyStatusCode.X509_V_ERR_UNABLE_TO_GET_CRL;
+                    }
+
+                    try
+                    {
+                        System.IO.Directory.CreateDirectory(ocspCache);
                     }
+                    catch
+                    {
+                        // Opportunistic create, suppress all errors.
+                    }
+
+                    return Interop.Crypto.X509ChainVerifyOcsp(_storeCtx, req, resp, ocspCache);
                 }
             }
+        }
+
+        private static string UrlPathAppend(string baseUri, ReadOnlyMemory<char> resource)
+        {
+            Debug.Assert(baseUri.Length > 0);
+            Debug.Assert(resource.Length > 0);
+
+            int count = baseUri.Length + resource.Length;
+            ValueTuple<string, ReadOnlyMemory<char>> state = ValueTuple.Create(baseUri, resource);
+
+            if (baseUri[baseUri.Length - 1] == '/')
+            {
+                return string.Create(
+                    count,
+                    state,
+                    (buf, st) =>
+                    {
+                        st.Item1.AsSpan().CopyTo(buf);
+                        st.Item2.Span.CopyTo(buf.Slice(st.Item1.Length));
+                    });
```

Comment:
Nit: you could instead write this as:
```C#
return string.Create(count, (baseUri, resource), (buf, st) =>
{
    st.baseUri.AsSpan().CopyTo(buf);
    st.resource.Span.CopyTo(buf.Slice(st.baseUri.Length));
});
```
avoiding both the direct use of ValueTuple as well as using Item1/Item2.